WebHostLib/options.py

```diff
@@ -69,7 +70,10 @@ def generate_game(player_name: str, formatted_options: dict):
             player_name: formatted_options,
         },
     }
-    r = requests.post("https://archipelago.gg/api/generate", json=payload)
+
+    url = urlparse(request.base_url)
+    port_string = f":{url.port}" if url.port else ""
+    r = requests.post(f"{url.scheme}://{url.hostname}{port_string}/api/generate", json=payload)
```

Comment:
this whole thing can just be replaced by url_for.
Also don't see  the point in doing a  self-refererantial post request  when we can just call  the function.

Reply:
`r = requests.post(url_for("api.generate_api"), json=payload)`
yields the error:
```
raise MissingSchema(
requests.exceptions.MissingSchema: Invalid URL '/api/generate': No scheme supplied. Perhaps you meant https:///api/generate?
```

I wish you would explain a little more rather than say "just call the function."

I tried to figure out what you meant by this, and the approach I thought of does not seem to work without modifying the `generate_api` function. The function requires POST data, which I do not know how to send via a function call. Is your intention for me to modify the function definition to accept data as an argument, and fall back to POST data if that argument contains `None`?

Reply:
In a similar time-frame to me explaining it, I can implement it, so I've done so in https://github.com/ArchipelagoMW/Archipelago/pull/3332 now.